abi/include/abi/errno.h

```diff
@@ -1,38 +1,117 @@
-#define ETIMEOUT            -13
-#define EINVAL              -14
-#define EBUSY               -15
-#define EOVERFLOW           -16
-#define EINTR               -17
-#define EMFILE              -18
-#define ENAMETOOLONG       -256
-#define EISDIR             -257
-#define ENOTDIR            -258
-#define ENOSPC             -259
-#define ENOTEMPTY          -261
-#define EBADF              -262
-#define EDOM               -263
-#define ERANGE             -264
-#define EXDEV              -265
-#define EIO                -266
-#define EMLINK             -267
-#define ENXIO              -268
-#define ENOFS              -269
-#define EBADCHECKSUM       -300
-#define ESTALL             -301
-#define EEMPTY             -302
-#define ENAK               -303
-#define EAGAIN           -11002
+#define EOK             __errno_t(      0)
+#define ENOENT          __errno_t(      1)
+#define ENOMEM          __errno_t(      2)
+#define ELIMIT          __errno_t(      3)
+#define EREFUSED        __errno_t(      4)
+#define EFORWARD        __errno_t(      5)
+#define EPERM           __errno_t(      6)
+#define EHANGUP         __errno_t(      7)
+#define EPARTY          __errno_t(      8)
+#define EEXIST          __errno_t(      9)
+#define EBADMEM         __errno_t(     10)
+#define ENOTSUP         __errno_t(     11)
+#define EADDRNOTAVAIL   __errno_t(     12)
+#define ETIMEOUT        __errno_t(     13)
+#define EINVAL          __errno_t(     14)
+#define EBUSY           __errno_t(     15)
+#define EOVERFLOW       __errno_t(     16)
+#define EINTR           __errno_t(     17)
+#define EMFILE          __errno_t(     18)
+#define ENAMETOOLONG    __errno_t(     19)
+#define EISDIR          __errno_t(     20)
+#define ENOTDIR         __errno_t(     21)
+#define ENOSPC          __errno_t(     22)
+#define ENOTEMPTY       __errno_t(     23)
+#define EBADF           __errno_t(     24)
+#define EDOM            __errno_t(     25)
+#define ERANGE          __errno_t(     26)
+#define EXDEV           __errno_t(     27)
+#define EIO             __errno_t(     28)
+#define EMLINK          __errno_t(     29)
+#define ENXIO           __errno_t(     30)
+#define ENOFS           __errno_t(     31)
+#define EBADCHECKSUM    __errno_t(     32)
+#define ESTALL          __errno_t(     33)
+#define EEMPTY          __errno_t(     34)
+#define ENAK            __errno_t(     35)
+#define EAGAIN          __errno_t(     36)
+
+/* libhttp error codes. Defining them here is a temporary hack. */
+
+#define HTTP_EMULTIPLE_HEADERS  __errno_t(  50)
+#define HTTP_EMISSING_HEADER    __errno_t(  51)
+#define HTTP_EPARSE             __errno_t(  52)
+
+/* libext4 error codes. Same as above. */
+
+#define EXT4_ERR_BAD_DX_DIR     __errno_t(  60)
+
+
+/* POSIX error codes + whatever nonstandard names crop up in third-party
+ * software. These are not used in HelenOS code, but are defined here in
+ * order to avoid nasty hacks in libposix.
+ *
+ * If you decide to use one of these in native HelenOS code,
```

Comment:
I think this split is not sustainable. People will simply start using those without moving them anywhere. Think of EPARTY, which was only supposed to be used by the kernel.

I also don't like that POSIX errnos are present in a HelenOS native header at all. It would be better if you could find a way to move these to libposix.